### PR TITLE
Integration Tests for CD-ROM UI [KNIP-520]

### PR DESCRIPTION
**Pull request by glekner**
Blocked by #2477 

File to review is `vm.detail.cdrom.ts`

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.cdrom.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          .all(by.css(editCdView.cdEjectBtn))
+          .then((ejects) => ejects.forEach((eject) => click(eject)));
+        await element
+          .all(by.cssContainingText('option', 'URL'))
+          .then((options) => click(options[0]));
+        await element
+          .all(by.cssContainingText('option', 'Persistent Volume Claim'))
+          .then((options) => click(options[1]));
+        await click(editCdView.saveButton());
+        await browser.wait(until.presenceOf(editCdView.diskSummary()));
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.cdrom.scenario.ts]: Maybe we should expect the PVC and URL CDs are displayed in the Overview page?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts [hunk 2]
```diff
@@ -166,13 +166,14 @@ export class VirtualMachine extends KubevirtDetailView {
...
 
-    await this.navigateToTab(TABS.OVERVIEW);
-    if (startOnCreation === true) {
-      // If startOnCreation is true, wait for VM to boot up
-      await vmView.waitForStatusIcon(vmView.statusIcons.running, VM_BOOTUP_TIMEOUT_SECS);
-    } else {
-      // Else wait for possible import to finish
-      await vmView.waitForStatusIcon(vmView.statusIcons.off, VM_IMPORT_TIMEOUT_SECS);
-    }
+    //   await this.navigateToTab(TABS.OVERVIEW);
```
Comment (glekner) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts]: waiting for @rhrazdil to fix this waiting for icons issue

**Comment by glekner** [comment]
updated tests and implemented @rhrazdil's suggestions. this is blocked until the waiting for the status icons issue is fixed

**Comment by glekner** [comment]
@rhrazdil can you review please?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.cdrom.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        await click(editCdView.cdAddBtn);
+        await click(editCdView.cdAddBtn);
+        await click(editCdView.saveButton);
+        await browser.wait(until.presenceOf(editCdView.diskSummary));
+
+        await vm.modalEditCDRoms();
+        await element
+          .all(by.css(editCdView.cdEjectBtn))
+          .then((ejects) => ejects.forEach((eject) => click(eject)));
+        await selectSelectorOption('#cd-rom-modal-select-type-cd-drive-1', 'url');
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.cdrom.scenario.ts]: these selectors would be nicer to define in the view file

**Comment by glekner** [comment]
implemented suggestions @rhrazdil

**Comment by rhrazdil** [comment]
lgtm, somebody else needs to approve

**Pull request merged**